src/hist.jl

```diff
@@ -186,56 +186,56 @@ binindex{T,N,E}(h::Histogram{T,N,E}, xs::NTuple{N,Real}) =
 end
 
 
-binvolume{T,E}(h::AbstractHistogram{T,1,E}, binidx::Integer) = binvolume(h, (binidx,))
-binvolume{V,T,E}(::Type{V}, h::AbstractHistogram{T,1,E}, binidx::Integer) = binvolume(V, h, (binidx,))
+binvolume(h::AbstractHistogram{T,1,E}, binidx::Integer) where {T,E} = binvolume(h, (binidx,))
+binvolume(::Type{V}, h::AbstractHistogram{T,1,E}, binidx::Integer) where {V,T,E} = binvolume(V, h, (binidx,))
 
-binvolume{T,N,E}(h::Histogram{T,N,E}, binidx::NTuple{N,Integer}) =
+binvolume(h::Histogram{T,N,E}, binidx::NTuple{N,Integer}) where {T,N,E} =
     binvolume(_promote_edge_types(h.edges), h, binidx)
 
-binvolume{V,T,N,E}(::Type{V}, h::Histogram{T,N,E}, binidx::NTuple{N,Integer}) =
+binvolume(::Type{V}, h::Histogram{T,N,E}, binidx::NTuple{N,Integer}) where {V,T,N,E} =
     prod(map((edge, i) -> _edge_binvolume(V, edge, i), h.edges, binidx))
 
-@inline _edge_binvolume{V}(::Type{V}, edge::AbstractVector, i::Integer) = V(edge[i+1]) - V(edge[i])
-@inline _edge_binvolume{V}(::Type{V}, edge::Range, i::Integer) = V(step(edge))
+@inline _edge_binvolume(::Type{V}, edge::AbstractVector, i::Integer) where {V} = V(edge[i+1]) - V(edge[i])
+@inline _edge_binvolume(::Type{V}, edge::Range, i::Integer) where {V} = V(step(edge))
 @inline _edge_binvolume(edge::AbstractVector, i::Integer) = _edge_binvolume(eltype(edge), edge, i)
 
 
-@inline _edges_nbins{N}(edges::NTuple{N,AbstractVector}) = map(_edge_nbins, edges)
+@inline _edges_nbins(edges::NTuple{N,AbstractVector}) where {N} = map(_edge_nbins, edges)
```

Comment:
While we're at it, I wonder whether this was intended to be `::Tuple{Vararg{AbsractVector}}`. IIRC since 0.6 `NTuple` implies homogeneous element types, but maybe our tests don't cover this. (No need to fix it in this PR though.)